app/models/ingredient.rb

```diff
@@ -1,3 +1,14 @@
 class Ingredient < ApplicationRecord
-   validates_presence_of :name, :cost
+  has_many :recipe_ingredients
+  has_many :recipes, through: :recipe_ingredients
+  validates_presence_of :name, :cost
+
+  def self.total_cost
+    @ingredients = Ingredient.all
+    cost = @ingredients.sum { |ingredient| ingredient.cost }
```

Comment:
I can understand your logic here, but this is a Ruby method and we want to rely on AR methods instead. The AR equivalent would be `.sum(:cost)`, where .sum is called on a collection of objects, and we pass in a symbol for the name of the attribute we want to sum. 

In addition, we can call this on `Ingredient.all`, or just `Ingredient` if we want to do it on the entire table. 

However, the user story calls for the total cost of the recipe by addnig up all of _that recipe's ingredients_, so this could work better and easier if we implemented this functionality in a Recipe instance method, like this: 
```ruby
def total_cost
   ingredients.sum(:cost) #where ingredients is the association on Recipe
end
```